utilities/tools/Test-NamePrefixAvailability.ps1

```diff
@@ -56,127 +64,164 @@ function Test-NamePrefixAvailability {
-
-        try {
-
-            # Extract Parameter Names
-            # -----------------------
-            $storageAccountFiles = $parameterFiles | Where-Object { $_ -match 'Microsoft.Storage/storageAccounts' }
-            $storageAccountNames = $storageAccountFiles | ForEach-Object { (ConvertFrom-Json (Get-Content $_ -Raw)).parameters.name.value } | Where-Object { $null -ne $_ }
-
-            $keyVaultFiles = $parameterFiles | Where-Object { $_ -match 'Microsoft.KeyVault/vaults' }
-            $keyVaultNames = $keyVaultFiles | ForEach-Object { (ConvertFrom-Json (Get-Content $_ -Raw)).parameters.name.value } | Where-Object { $null -ne $_ }
-
-            $acrFiles = $parameterFiles | Where-Object { $_ -match 'Microsoft.ContainerRegistry/registries' }
-            $acrNames = $acrFiles | ForEach-Object { (ConvertFrom-Json (Get-Content $_ -Raw)).parameters.name.value } | Where-Object { $null -ne $_ }
+        foreach ($parameterFile in $parameterFiles) {
+
+            foreach ($relevantResourceType in $relevantResourceTypes) {
+                switch ($relevantResourceType) {
+                    'Microsoft.Storage/storageAccounts' {
+                        $filter = 'storageAccountName:'
+                    }
+                    'Microsoft.ContainerRegistry/registries' {
+                        $filter = 'registryName:'
+                    }
+                    'Microsoft.KeyVault/vaults' {
+                        $filter = 'keyVaultName:'
+                    }
+                    Default {
+                        Write-Warning "The entered resource Type '$_' could not be matched."
+                        continue
+                    }
+                }
+            
+                $temp = $null
 
-            $subscriptionId = (Get-AzContext).Subscription.Id
+                # determine if entry is of one of the resourceTypes using the filter variable
+                $temp = Get-Content -Path $parameterFile | ForEach-Object {
+                    if ($_ -match "$filter\s'") { $_ }
+                }
 
-            # Storage
-            Write-Host "`nCheck Storage Accounts" -ForegroundColor 'Cyan'
-            Write-Host '======================' -ForegroundColor 'Cyan'
-            foreach ($storageAccountName in $storageAccountNames) {
-                $path = '/subscriptions/{0}/providers/Microsoft.Storage/checkNameAvailability?api-version=2021-04-01' -f $subscriptionId
-                $requestInputObject = @{
-                    Method  = 'POST'
-                    Path    = $path
-                    Payload = @{
-                        name = $storageAccountName
-                        type = 'Microsoft.Storage/storageAccounts'
-                    } | ConvertTo-Json
+                # determine serviceshort default value if no parameter has been supplied
+                if (!$overwrittenServiceShort) {
+                    $serviceShort = Get-Content -Path $parameterFile | ForEach-Object {
```

Comment:
The whole block can be shortened via
```PowerShell
// With pre-loaded file content array
$serviceShort = $fileContent | ForEach-Object { 
if ($_ -Match "^param serviceShort string = '(\w+)'$") {
    $matches[1]
    break 
}
```

Reply:
@AlexanderSehr Updated with your suggestion